include/array/array.h

```diff
@@ -202,6 +202,16 @@ class index_iterator {
     ++i_;
     return *this;
   }
+  NDARRAY_INLINE NDARRAY_HOST_DEVICE index_iterator& operator+=(index_t r) {
```

Comment:
In general I'm avoiding relying on the compiler inlining trivial functions too much. Any one case is of course negligible, but I think they added up to make debug builds excruciatingly slow. At some point a while back I spent some time trying to reduce the pre-optimization code size, to help with debug builds and build time. I didn't do it in the other case you are referring to (for consistency) because that `+=` was non-trivial.

Reply:
I might have misunderstood your comment... I'll add the other operators

Reply:
I'm just going to implement the full set of stuff required by https://en.cppreference.com/w/cpp/iterator/random_access_iterator